src/users/user.php

```diff
@@ -0,0 +1,371 @@
+
+					return;
+				}
+			}
+		}
+	}
+
+	/**
+	 * Function get for user record.
+	 *
+	 * @return object|void User details on success otherwise raise error
+	 *
+	 * @since   2.0
+	 */
+	public function get()
+	{
+		$input = JFactory::getApplication()->input;
+		$id = $input->get('id', 0, 'int');
+
+		/*
+		 * If we have an id try to fetch the user
+		 * @TODO write user field mapping logic here
+		 */
+		if ($id)
+		{
+			$user = JUser::getInstance($id);
+
+			if (! $user->id)
+			{
+				ApiError::raiseError(400, JText::_('PLG_API_USERS_USER_NOT_FOUND_MESSAGE'));
+
+				return;
+			}
+
+			$this->plugin->setResponse($user);
+		}
+		else
+		{
+			$user = JFactory::getUser();
+
+			if ($user->guest)
+			{
+				ApiError::raiseError(400, JText::_('JERROR_ALERTNOAUTHOR'));
+			}
+
+			$this->plugin->setResponse($user);
+		}
+	}
+
+	/**
+	 * Function to returns userid if a user exists depending on email
+	 *
+	 * @param   string  $email  The email to search on.
+	 *
+	 * @return  integer  The user id or 0 if not found.
+	 *
+	 * @since   2.0
+	 */
+	private function getUserId($email)
+	{
+		// Initialise some variables
+		$db		= JFactory::getDbo();
+		$query	= $db->getQuery(true)
+			->select($db->quoteName('id'))
+			->from($db->quoteName('#__users'))
+			->where($db->quoteName('email') . ' = ' . $db->quote($email));
+		$db->setQuery($query, 0, 1);
+
+		return $db->loadResult();
+	}
+
+	/**
+	 * Funtion for bind and save data and return response.
+	 *
+	 * @param   Object   $user      The user object.
+	 * @param   Array    $formData  Array of user data to be added or updated.
+	 * @param   Boolean  $flag      Flag to differnciate the update of create action.
+	 *
+	 * @return  object|void  $response  the response object created on after user saving. void and raise error
+	 *
+	 * @since   2.0
+	 */
+	private function storeUser($user, $formData, $flag = 0)
+	{
+		$response = new stdClass;
+
+		if (!$user->bind($formData))
+		{
+			ApiError::raiseError(400, $user->getError());
+
+			return;
+		}
+
+		if (!$user->save())
+		{
+			ApiError::raiseError(400, $user->getError());
+
+			return;
+		}
+
+		$response->id = $user->id;
+
+		if ($flag)
+		{
+			$response->message = JText::_('PLG_API_USERS_ACCOUNT_UPDATED_SUCCESSFULLY_MESSAGE');
+		}
+		else
+		{
+			$response->message = JText::_('PLG_API_USERS_ACCOUNT_CREATED_SUCCESSFULLY_MESSAGE');
+		}
+
+		return $response;
+	}
+
+	/**
+	 * Function delete is used to delete the respective user record.
+	 *
+	 * @return void
+	 *
+	 * @since   2.0
+	 */
+	public function delete()
+	{
+		$app = JFactory::getApplication();
+		$userIdentifier	= $app->input->get('id', 0, 'STRING');
+		$xidentifier	= $app->input->server->get('HTTP_IDENTIFIER');
+
+		$loggedUser  = JFactory::getUser();
+
+		// Check if I am a Super Admin
+		$iAmSuperAdmin = $loggedUser->authorise('core.admin');
+
+		$userToDelete = $this->retriveUser($xidentifier, $userIdentifier);
+
+		if ($userToDelete->id)
```

Comment:
handle else first, move code out of `if`

Reply:
Done